packages/button/addon/components/nucleus-button.js

```diff
@@ -426,7 +426,9 @@ class NucleusButton extends Component {
         })
         .finally(() => {
           run.later(() => {
-            set(this, '_buttonState', BUTTON_STATE.DEFAULT)
+            if (!this.isDestroyed) {
+              set(this, '_buttonState', BUTTON_STATE.DEFAULT)
+            }
```

Comment:
Instead of growing the guard checks everywhere, I'd suggest you opt for `ember-concurrency`. Thoughts?

Reply:
We wanted to refactor the async part using ember-concurrency, but unfortunately didn't have enough bandwidth. PRs welcome :)

Reply:
Neither do I :( I was thinking these components were being built ground-up 👨‍🎨